### PR TITLE
Fix PIF carrier passthrough issues

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

**Comment by robhoes** [comment]
Note that much of the first patch is to resolve a circular dependency between modules. The main point of the patch is the bit in xapi_xenops.ml.

File: ocaml/xapi/monitor_dbcalls_cache.ml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+open Threadext
```
Comment (simonjbeaumont) [inline ocaml/xapi/monitor_dbcalls_cache.ml]: This file is missing the standard Citrix spiel.

**Comment by simonjbeaumont** [comment]
Looks good. Once https://github.com/xapi-project/xen-api/pull/2237#discussion_r31024849 has been addressed this is good to go. No need for a re-review.

**Comment by robhoes** [comment]
@simonjbeaumont Thanks for the review. I've added the spiel.

**Pull request merged**